channel_manager.ts

```diff
@@ -54,7 +56,8 @@ export class ChannelManager {
       throw Error(`Unable to join channel(s). Channel(s) either do not exist or aren't joinable.`);
     }
 
-    await member.roles.add(roles);
+    this.bot.LogTrace(`preparing to add roles: ${formatJsonMessage(roles)}`);
+    member = await member.roles.add(roles);
```

Comment:
I've added this to both the join and leave functions for 'correctness'.
roles.add(...) returns a promise, but that value is being discarded.
There's a chance that the promise isn't correctly being awaited and could be causing problems.

This updates the member variable with the return value -- which should be the most correct, and also forces the code to actually await on this Promise as member is used later in the code.